### DISCUSSION TITLE
Now that we have a mushroom select card ...

### DISCUSSION THREAD

**Discussion by kbrown01** [piitaya/lovelace-mushroom#966] [Ideas]
How about adding a source selection to the mushroom media card so that we can select the source?????? 

Using mini-media:

![image](https://user-images.githubusercontent.com/723363/213804955-dfe29dc4-4f8b-4cde-82d4-864aeed91532.png)

Now if I could add the select to this:

![image](https://user-images.githubusercontent.com/723363/213805102-0f575a15-1a76-4813-8b1d-47fb55d99923.png)

I would be in heaven!

And no I don't want it on multiple lines ... nice, compact like mini-media and right inline. Reason? Because that whole card is done with auto-entities and it would be a pain to implement all that repeat code just for an input select for sources. Right now, I have this for mini-media:

`              - type: custom:auto-entities
                card:
                  type: entities
                  title: Speaker Zones
                  show_header_toggle: false
                  state_color: true
                  group: true
                  hide:
                    play_pause: true
                filter:
                  include:
                    - entity_id: media_player.xan*
                      options:
                        sourceicon: mdi:music
                        type: custom:mini-media-player
                        group: true
                        hide:
                          controls: false
                          icon: true
                          source: false
                  exclude:
                    - entity_id: media_player.xantech8_xantech_house_audio
`

You can see the option hide source is false. Would it not be so great to have:

`type: custom:auto-entities
card:
  type: grid
  columns: 1
  title: Speaker Zones
  show_header_toggle: false
  state_color: true
  square: false
card_param: cards
filter:
  include:
    - entity_id: media_player.xan*
      options:
        type: custom:mushroom-media-player-card
        layout: horizontal
        fill-container: true
        media_controls:
          - play_pause_stop
        volume_controls:
          - volume_mute
          - volume_set
        show_volume_level: true
        **show_source_selection: true**
        collapsible_controls: true
  exclude:
    - entity_id: media_player.xantech8_xantech_house_audio
`

Add entry for "show_source_selection" or something?

**Comment by kbrown01** [comment]
To give you an idea, I have converted much of of home to mushroom --- I love it! But this GUI remains:

![image](https://user-images.githubusercontent.com/723363/213807638-937fe1fd-f3b5-4f21-978f-b6fb39ef8f5b.png)

Which is the most used panel in the house. The music center. I could convert the input selections but woould still have the mini-media players only because of this. The rest can be chips. It would be so great.

**Comment by fruhbolk** [comment]
Any luck with this? I got the Monoprice 6 Zone amp and I've copied some of your setup. I'm running into the same exact issue.
     _Pretty mushroom card and have an extra couple clicks/taps per source change or stick with the less aesthetic mini-media player?_

**Comment by kbrown01** [comment]
Not really-ish:

```
                            type: custom:auto-entities
                            card:
                              type: vertical-stack
                              show_header_toggle: false
                              state_color: true
                              square: false
                            card_param: cards
                            filter:
                              template: >
                                {% for MEDIAPLAYER in states.media_player |
                                selectattr("entity_id","in",integration_entities("xantech"))
                                -%}
                                  {% set SELECTZONE = "input_select.dax_source_" + (MEDIAPLAYER.entity_id).split("xantech8_")[1]
                                -%}
                                  {% if MEDIAPLAYER.entity_id != "media_player.xantech8_xantech_house_audio"
                                -%}
                                    {{
                                      {
                                        'type': 'custom:layout-card',
                                        'layout_type': 'custom:grid-layout',
                                        'layout': {
                                          'max_cols': 3,
                                          'grid-template-columns': 'auto 150px'
                                        },
                                        'cards': [
                                        {
                                          'type': 'custom:decluttering-card',
                                          'template': 'mediaplayer',
                                          'variables': [
                                            {
                                             'mediaplayer': MEDIAPLAYER.entity_id
                                            }
                                          ]
                                        },
                                        {
                                            "type": "custom:mushroom-select-card",
                                            "entity": SELECTZONE,
                                            "name": "Set Zone Sources",
                                            "layout": "horizontal",
                                            "icon_type": "none",
                                            "primary_info": "none",
                                            "secondary_info": "none",
                                            "card_mod": {"style": "ha-card {box-shadow:  none;}"}
                                        }
                                      ]}
                                    }},
                                  {%- endif %}
                                {%- endfor %}
```

![image](https://github.com/user-attachments/assets/1551d533-ff93-44be-96de-dca73a4bb4be)

**Reply by kbrown01** [reply]
So I use multiple cards, media-player and select. At least all mushroom but have automation(s) happen on changing the source.

**Reply by fruhbolk** [reply]
Not bad! That is the route I was starting to consider. I like your new tab set-up too. Any chance you have your full updated yaml posted for that page? I originally went off the one posted [here](https://github.com/rsnodgrass/hass-xantech?tab=readme-ov-file).

**Reply by kbrown01** [reply]
Well, your question prompted me to take another look. 
I am now re0-writing in bubble and taking out mushroom. You can put the input select right inside and give a more professional look.
Just tested and stating to change at least the media players.

![image](https://github.com/user-attachments/assets/fb7a4673-1ea6-4665-98cf-376a70403413)

**Reply by kbrown01** [reply]
Like this?

![image](https://github.com/user-attachments/assets/669e88fa-91fe-4de7-8caf-5d143ed97293)

Use bubble-card and sub card(s) you can put the source-list in the card, but you would need script to ;ick up changes and do something. My card uses decluttering but this is the template that draws a card:

```
  mediaplayer_bubble:
    default:
      - mediaplayer: media_player.xantech8_deck
    card:
      type: custom:bubble-card
      card_type: button
      entity: '[[mediaplayer]]'
      tap_action:
        action: call-service
        service: media_player.toggle
        target:
          entity_id: '[[mediaplayer]]'
      layout: horizontal
      show_state: false
      button_type: slider
      show_attribute: true
      attribute: volume_level
      hide:
        play_pause_button: true
        volume_button: true
        previous_button: true
        next_button: true
        power_button: true
      sub_button:
        - entity: '[[selectzone]]'
          attribute: source_list
          show_state: true
          show_icon: false
```

This draws them:

```
                            type: custom:auto-entities
                            card:
                              type: vertical-stack
                            card_param: cards
                            filter:
                              template: >
                                {% for MEDIAPLAYER in states.media_player |
                                selectattr("entity_id","in",integration_entities("xantech"))
                                -%}
                                  {% set SELECTZONE = "input_select.dax_source_" + (MEDIAPLAYER.entity_id).split("xantech8_")[1]
                                -%}
                                  {% if MEDIAPLAYER.entity_id != "media_player.xantech8_xantech_house_audio"
                                -%}
                                    {{
                                      {
                                        'type': 'vertical-stack',
                                        'cards': [
                                        {
                                          'type': 'custom:decluttering-card',
                                          'template': 'mediaplayer_bubble',
                                          'variables': [
                                               {'mediaplayer': MEDIAPLAYER.entity_id},
                                               {'selectzone': SELECTZONE}
                                          ]
                                        }
                                      ]}
                                    }},
                                  {%- endif %}
                                {%- endfor %}
```

**Reply by fruhbolk** [reply]
Looks great! Thanks for the update.

Here's what I've got so far:
![image](https://github.com/user-attachments/assets/6f73f55f-9e87-4c77-ab93-fa4f73fbc599)
I played with the border radius to better match mushroom cards since that's primarily what I'm using elsewhere. This is the draft for my simple mobile dashboard. The top media player is my main streamer (plugged into my Monoprice amp) running through Music Assistant. I'll probably play around with something a bit more robust at some point to use in desktop or with a dedicated tablet. I like your tabs and grouping system for that.

**Reply by kbrown01** [reply]
Looks awesome. On my "music source" tab to run audio streams, I am adding artist, track, album.

I like it that way instead of overlaying album art.

![image](https://github.com/user-attachments/assets/696af377-c5b7-4c87-a7b5-a49d6b8da74e)

Used the same technique for the Pandora. I run that through Alexa, you select a genre or artist and it send the command to Alexa:

"Play Pop music from Pandora on Stereo Alexa"

![image](https://github.com/user-attachments/assets/85532f94-e874-4cdd-93f1-766cde30705e)
